src/main/java/selenide/page/object/POProvider.java

```diff
@@ -0,0 +1,35 @@
+package selenide.page.object;
+
+public class POProvider {
```

Comment:
No need singleton here. You can directly init Pages in Actions. Or use Factory in case you have specific logic for page initialization